python/lsst/daf/butler/core/datastore.py

```diff
@@ -98,6 +160,24 @@ def __init__(self, config, registry):
         self.config = DatastoreConfig(config)
         self.registry = registry
         self.name = "ABCDataStore"
+        self._transaction = None
+
+    @contextlib.contextmanager
+    def transaction(self):
+        """Context manager supporting `Datastore` transactions.
+
+        Transactions can be nested, and are to be used in combination with
+        `Registry.transaction`.
+        """
+        self._transaction = DatastoreTransaction(self._transaction)
+        try:
+            yield
+        except BaseException:
+            self._transaction.rollback()
+            raise
+        finally:
```

Comment:
I know I usually get confused by this, but shouldn't this be `else:` rather than `finally`? You don't want the commit to be called after the rollback has run, you only want the commit to happen if the except did not happen.